compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -529,6 +529,32 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             obligation.predicate.to_poly_trait_ref(),
             trait_ref,
         )?);
+
+        let lang_items = self.tcx().lang_items();
+        if [lang_items.fn_once_trait(), lang_items.fn_trait(), lang_items.fn_mut_trait()]
+            .contains(&Some(obligation.predicate.def_id()))
+            // Skip `MiscObligation`s for better output.
+            && matches!(obligation.cause.code, ObligationCauseCode::BindingObligation(_, _))
+        {
+            // Do not allow `foo::<fn() -> A>();` for `A: !Sized` (#82633)
+            let fn_sig = obligation.predicate.self_ty().skip_binder().fn_sig(self.tcx());
+            let ty = self.infcx.replace_bound_vars_with_placeholders(fn_sig.output());
```

Comment:
Using `replace_bound_vars_with_placeholders` pacified CI, but I still don't understand why it wasn't triggering on my machine. Maybe the `assert!` wasn't being run locally?